src/databricks/labs/ucx/hive_metastore/locations.py

```diff
@@ -113,6 +127,71 @@ def _try_fetch(self) -> Iterable[ExternalLocation]:
         for row in self._fetch(f"SELECT * FROM {self._schema}.{self._table}"):
             yield ExternalLocation(*row)
 
+    def _get_ext_location_definitions(self, missing_locations: list[ExternalLocation]) -> list:
+        tf_script = []
+        cnt = 1
+        for loc in missing_locations:
+            if loc.location.startswith("s3://"):
+                res_name = loc.location[5:].replace("/", "_")[:-1]
+            elif loc.location.startswith("gcs://"):
+                res_name = loc.location[6:].replace("/", "_")[:-1]
+            elif loc.location.startswith("abfss://"):
+                res_name = (
+                    loc.location[loc.location.index("@") + 1 :]
+                    .replace(".dfs.core.windows.net", "")
+                    .replace("/", "_")[:-1]
+                )
+            else:
+                res_name = f"name_{cnt}"
+            script = f'resource "databricks_external_location" "{res_name}" {{ \n'
+            script += f'    name = "{res_name}"\n'
+            script += f'    url  = "{loc.location[:-1]}"\n'
+            script += "    credential_name = databricks_storage_credential.<storage_credential_reference>.id\n"
+            script += "}\n"
+            tf_script.append(script)
+            cnt += 1
+        return tf_script
+
+    def _match_table_external_locations(self) -> tuple[list, list[ExternalLocation]]:
```

Comment:
what's the type of the first return argument in a tuple?..

Reply:
its a combination of str and count. name from ws.external location and table count from ExternalLocation

Reply:
Why isn't this clear from a type?